core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3033,9 +3033,16 @@ private void setBuildsAndWorkspacesDir() throws IOException, InvalidBuildsDir {
         }
 
         String newWorkspacesDir = SystemProperties.getString(WORKSPACES_DIR_PROP);
-        if (newWorkspacesDir != null && !workspaceDir.equals(newWorkspacesDir)) {
-            LOGGER.log(Level.WARNING, "Changing workspaces directories from {0} to {1}. Beware that no automated data migration will occur.",
-                       new String[]{workspaceDir, newWorkspacesDir});
+        if (newWorkspacesDir != null) {
+        	Level level;
+        	if(!workspaceDir.equals(newWorkspacesDir)) {
+        		level = Level.INFO;
+        	}
+        	else {
+        		level = Level.WARNING;
```

Comment:
won't this cause the WARNING message to always be logged on all startups except the first one?